cpp/ql/src/semmle/code/cpp/models/interfaces/FunctionInputsAndOutputs.qll

```diff
@@ -35,73 +117,178 @@ class InParameter extends FunctionInput, TInParameter {
+
+  /**
+   * Holds if this is the value returned by a function.
+   * Example:
+   * ```
+   * int getInt();
+   * char* getPointer();
+   * float& getReference();
+   * ```
+   * `isReturnValue()` holds for the `FunctionOutput` that represents the value returned by
+   *   `getInt()` (with type `int`).
+   * `isReturnValue()` holds for the `FunctionOutput` that represents the value returned by
+   *   `getPointer()` (with type `char*`).
+   * `isReturnValue()` holds for the `FunctionOutput` that represents the "value" of the reference
+   *   returned by `getReference()` (with type `float&`), _not_ the value of the referred-to
+   *   `float`.
+   */
+  predicate isReturnValue() { none() }
+
+  /**
+   * Holds if this is the value returned by a function.
+   * DEPRECATED: Use `isReturnValue()` instead.
+   */
+  deprecated final predicate isOutReturnValue() { isReturnValue() }
+
+  /**
+   * Holds if this is the output value pointed to by the return value of a function, if the function
+   *   returns a pointer, or the output value referred to by the return value of a function, if the
+   *   function returns a reference.
+   * Example:
+   * ```
+   * char* getPointer();
+   * float& getReference();
+   * int getInt();
+   * ```
+   * `isReturnValueDeref()` holds for the `FunctionOutput` that represents the value of
+   *   `*getPointer()` (with type `char`).
+   * `isReturnValueDeref()` holds for the `FunctionOutput` that represents the value of
+   *   `getReference()` (with type `float`).
+   * There is no `FunctionOutput` of `getInt()` for which `isReturnValueDeref()` holds because the
+   *   return type of `getInt()` is neither a pointer nor a reference.
+   */
+  predicate isReturnValueDeref() { none() }
+
+  /*
+   * Holds if this is the output value pointed to by the return value of a function, if the function
+   *   returns a pointer, or the output value referred to by the return value of a function, if the
+   *   function returns a reference.
+   * DEPRECATED: Use `isReturnValueDeref()` instead.
+   */
+
+  deprecated final predicate isOutReturnPointer() { isReturnValueDeref() }
 }
 
-class OutParameterPointer extends FunctionOutput, TOutParameterPointer {
+class OutParameterDeref extends FunctionOutput, TOutParameterDeref {
   ParameterIndex index;
 
-  OutParameterPointer() { this = TOutParameterPointer(index) }
+  OutParameterDeref() { this = TOutParameterDeref(index) }
 
-  override string toString() { result = "OutParameterPointer " + index.toString() }
+  override string toString() { result = "OutParameterDeref " + index.toString() }
 
   ParameterIndex getIndex() { result = index }
 
-  override predicate isOutParameterPointer(ParameterIndex i) { i = index }
+  override predicate isParameterDeref(ParameterIndex i) { i = index }
 }
 
-class OutQualifier extends FunctionOutput, TOutQualifier {
+class OutQualifierObject extends FunctionOutput, TOutQualifierObject {
   override string toString() { result = "OutQualifier" }
```

Comment:
```suggestion
  override string toString() { result = "OutQualifierObject" }
```

Reply:
I'm not sure what's going on with the indentation in this edit suggestion. I didn't suggest any change of indentation.

Reply:
Fixed